data/campaigns/Eastern_Invasion/scenarios/06b_Soradoc.cfg

```diff
@@ -0,0 +1,1039 @@
+                    [/have_unit]
+                    [variable]
+                        name=met_yannic
+                        not_equals=yes
+                    [/variable]
+                [/show_if]
+            [/objective]
+            [objective]
+                {OPTIONAL_OBJECTIVE_CAPTION}
+                description= _ "Keep Yannic alive (bonus reward)."
+                condition=win
+                [show_if]
+                    [have_unit]
+                        id=Yannic
+                    [/have_unit]
+                    [variable]
+                        name=met_yannic
+                        equals=yes
+                    [/variable]
+                [/show_if]
+            [/objective]
+            [objective]
+                description= _ "Death of Gweddry, Dacyn, or Owaec"
+                condition=lose
+            [/objective]
+            {TURNS_RUN_OUT}
+            [gold_carryover]
+                bonus=no
+                carryover_percentage=40
+            [/gold_carryover]
+        [/objectives]
+
+        {RECALL_XY (Dacyn) 22 31}
+        {RECALL_XY (Owaec) 23 31} # make it more difficult for him to rush across the river
+        {RECALL_XY (Addogin) 23 30}
+        {RECALL_XY (Terraent) 23 30}
+        {RECALL_XY (Hahid al-Ali) 23 30}
+    [/event]
+
+    [event]
+        name=start
+
+        [message]
+            speaker=Gweddry
+            message= _ "The air is filled with smoke. I fear the town is already lost."
+        [/message]
+        [message]
+            speaker=Owaec
+            message= _ "Madness! How can the city have fallen? The garrison is stout-hearted and loyal to a fault. I know many of them personally. They would hold the city to their dying breaths."
+        [/message]
+        [message]
+            speaker=Dacyn
+            message= _ "Were you expecting a victory parade? Soradoc will be the first of many if we cannot accomplish our goal."
+        [/message]
+        [message]
+            speaker=Gweddry
+            message= _ "Wait, look, the flag of Wesnoth still flies over the western barracks!"
+        [/message]
+
+        {SCROLL_TO 15 14}
+        [delay]
+            time=100
+        [/delay]
+
+        [message]
+            speaker=Yannic
+            message= _ "Are you our reinforcements? Weldyn promised us aid days ago! The undead already control most of the city. We’re hiding our wounded in the temple, but if we lose the western quarter as well, I fear that no one will escape."
+        [/message]
+        [message]
+            speaker=Owaec
+            message= _ "By the clans, some of them yet live! We must move quickly if we are to rescue the garrison! If we can defeat this wave of undead, they may yet have time to evacuate."
+        [/message]
+        [message]
+            speaker=Gweddry
+            message= _ "I agree, but... there are many powerful undead sieging the city. Do we have the strength to defeat them all?"
```

Comment:
Either 'powerful undead besieging the city' or 'powerful undead laying siege to the city'.